src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.DSA.Shared.cs

```diff
@@ -276,9 +240,13 @@ internal static DSAParameters ToDSAParameters(this byte[] cspBlob, bool includeP
 
                         if (includePrivateParameters)
                         {
-                            // Since DSSPUBKEY key is used for both public and private keys, we got X
-                            // but not Y. To get Y, do another export and ask for public key blob.
-                            byte[] cspPublicBlob = ExportKeyBlob(false, safeKeyHandle);
+                            // Since DSSPUBKEY is used for either public or private key, we got X
+                            // but not Y. To get Y, use the public key blob.
+
+                            if (cspPublicBlob == null)
+                            {
+                                throw new PlatformNotSupportedException(); // Edge case with Unix interop
```

Comment:
After further thought, I think this should be CryptographicUnexpectedOperationException instead. In theory a similar scenario can exist in Windows (but not in this code path) and it will throw some sort of WindowsCryptographicException. So at least we will have the same CryptographicException base class. Added comment to the code, but not a string to the message.